### PR TITLE
No Dell Java please.

### PR DESCRIPTION
**Pull request by mick-t**
This update removes dell java packages on existing installations and on new installation will not install the java related packages.

These are the packages that will no longer be installed:

- 'srvadmin-jre', 'srvadmin-all', 'srvadmin-tomcat', 'srvadmin-webserver'

To override this setting, set the following to true:

`dsu::srvadmin_install_all=true`

We don't install Dell's bundled java as it shows up on our security dashboards and has not been updated since mid-2022. We don't uses the related the tomcat and webserver packages.

### CODE REVIEW

**Review by mathisjames — APPROVED** [review]

File: manifests/params.pp [hunk 1]
```diff
@@ -7,8 +7,20 @@
   $package_version = 'latest'
   $srvadmin = true
   $srvadmin_package = 'srvadmin-all'
+  $srvadmin_packages_java = ['srvadmin-all', 'srvadmin-webserver', 'srvadmin-tomcat', 'srvadmin-jre', ]
```
Comment (bp85) [inline manifests/params.pp:10]: have you tested removing `srvadmin-all` doesn't remove all the dependencies?

File: manifests/params.pp [hunk 2]
```diff
@@ -7,8 +7,20 @@
...
+  $srvadmin_packages_no_java = ['srvadmin-argtable2', 'srvadmin-base', 'srvadmin-deng', 'srvadmin-deng-snmp', 'srvadmin-hapi',
```
Comment (bp85) [inline manifests/params.pp:11]: I'm not sure how often this list gets updated, but we should probably look at alternate way

File: manifests/params.pp (hunk 1 above)
Comment (mick-t) [reply manifests/params.pp:10]: Yes I have. Removing `srvadmin-all` only removes that package, installing `srvadmin-all` installs all the other packages PLUS the java related packages.

File: manifests/params.pp (hunk 2 above)
Comment (mick-t) [reply manifests/params.pp:11]: Yes you're correct. I'm not sure how we can do that without looking at the source of any updated `srvadmin-all` packages. Plus any new versions would require a change. 

What we can do is overwrite the package list with:

`dsu:: srvadmin_packages_no_java = '['package0', 'package1', ...]`

At the moment it's a stop gap to get rid of the java packages.

**Pull request merged**